### PR TITLE
Add SBX environment

### PR DESCRIPTION
**Pull request by jose-gataca**

### CODE REVIEW

File: admin/controller/catalog.go [hunk 1]
```diff
@@ -40,24 +40,31 @@ func (ch *catalogHandler) getPresentationCatalog(c echo.Context) error {
...
+	if err != nil {
+		return c.JSON(getStatusCode(err), ResponseError{Message: err.Error()})
 	}
-	data, err := (*ch.cpService).GetCatalogByCredentials(c, did)
+	sbx, err := (*ch.cpService).GetCatalogByCredentials(c, did, true)
 	if err != nil {
 		return c.JSON(getStatusCode(err), ResponseError{Message: err.Error()})
 	}
-	return c.JSON(http.StatusOK, data)
+	return c.JSON(http.StatusOK, map[string]interface{}{"regular": regular, "sbx": sbx})
```
Comment (samu-gataca) [inline admin/controller/catalog.go:61]: This is to manage the catalog? or to provide the catalog to users?

File: admin/service/catalogService.go [hunk 2]
```diff
@@ -31,24 +32,32 @@ func NewCatalogService(catalogDao *admin.CatalogDao, eventer event.Dispatcher) a
...
 
-func (cs *catalogService) GetByKey(ctx echo.Context, key string) ([]models.Resource, error) {
-
-	cl, err := (*cs.catalogDao).GetByKey(key)
+func (cs *catalogService) GetByKey(ctx echo.Context, key string, sbx bool) ([]models.Resource, error) {
+	sbxKey := key
+	if sbx && !strings.HasSuffix(key, models.SandboxSuffix) {
+		sbxKey += models.SandboxSuffix
+	}
+	cl, err := (*cs.catalogDao).GetByKey(sbxKey)
```
Comment (samu-gataca) [inline admin/service/catalogService.go:40]: WTF?

File: admin/service/gatacaLoginService.go [hunk 3]
```diff
@@ -329,6 +329,7 @@ func (ls *gatacaLoginService) generateTokenForUser(ctx echo.Context, dbuser *mod
 	claims := &models.CustomClaims{
 		Permissions: []string{dbuser.Role},
 		Tenants:     dbuser.Tenants,
+		Sandbox:     dbuser.Sandbox,
```
Comment (samu-gataca) [inline admin/service/gatacaLoginService.go:332]: Nice

File: models/catalog.go [hunk 4]
```diff
@@ -49,6 +50,7 @@ type Issuer struct {
 type Catalog struct {
 	Key       string       `gorm:"primary_key" json:"key"`
 	Value     string       `gorm:"type: json; not null" json:"value"`
+	Sbx       bool         `gorm:"" json:"Sbx"`
```
Comment (samu-gataca) [inline models/catalog.go:53]: All the catalog is to SBX, not just some entries?

File: models/config.go [hunk 5]
```diff
@@ -50,6 +50,7 @@ type TokenConfig struct {
 	PrivateKeyData  string `gorm:"" json:"privateKeyData"` //Do not expose private keys on API!
 	PrivateKeyType  string `gorm:"" json:"privateKeyType"`
 	Ledger          string `gorm:"not_null" json:"ledger"`
+	Sbx             bool   `gorm:"" json:"sandbox"`
```
Comment (samu-gataca) [inline models/config.go:53]: Why is important a DID to SBX and not?

File: models/users.go [hunk 6]
```diff
@@ -86,6 +87,7 @@ type User struct {
 	Salt      string         `gorm:"not null" json:"-"`
 	Role      string         `gorm:"not null" json:"role"`
 	Status    string         `gorm:"not null" json:"status"`
+	Sandbox   bool           `gorm:"" json:"sandbox"`
```
Comment (samu-gataca) [inline models/users.go:90]: Users need to know if they are from SBX? If they are in the catalog as a SBX it's enough, no?

**Review by samu-gataca — COMMENTED** [review]
Same comments Backbone PR. There is too much impact on the functionality than I would like to have. Because it's an "external" feature. It's not a functional requirement from the functionality. It's a functional requirement from GATACA to make "testing" on a prod environment.
Technically it's possible as you build, but I prefer to discuss better before to approve. 
My advise is to use the context (already expanded to all the methods) to store the SBX info, and then use it when it's necessary. What do you think?

**Comment by jose-gataca** [comment]
> Same comments Backbone PR. There is too much impact on the functionality than I would like to have. Because it's an "external" feature. It's not a functional requirement from the functionality. It's a functional requirement from GATACA to make "testing" on a prod environment.
> Technically it's possible as you build, but I prefer to discuss better before to approve.
> My advise is to use the context (already expanded to all the methods) to store the SBX info, and then use it when it's necessary. What do you think?

I agree, I would say that there are much more changes than "impact". I am testing it right now, so it's not ready to merge. From what we discussed, I am checking if it is needed to have users as Sandbox or not.

**Review by samu-gataca — APPROVED** [review]

**Pull request merged**